test/e2e/native_histograms_test.go

```diff
@@ -38,35 +45,53 @@ func TestQueryNativeHistograms(t *testing.T) {
+	testutil.Ok(t, err)
+	_, err = writeHistograms(ctx, now, histograms, rawRemoteWriteURL2)
+	testutil.Ok(t, err)
+
+	ts := func() time.Time { return now }
 
 	// Make sure we can query histogram from both Prometheus instances.
-	queryAndAssert(t, ctx, prom1.Endpoint("http"), func() string { return "fake_histogram" }, time.Now, promclient.QueryOptions{}, expectedHistogramModelVector(nil))
-	queryAndAssert(t, ctx, prom2.Endpoint("http"), func() string { return "fake_histogram" }, time.Now, promclient.QueryOptions{}, expectedHistogramModelVector(nil))
+	queryAndAssert(t, ctx, prom1.Endpoint("http"), func() string { return testHistogramMetricName }, ts, promclient.QueryOptions{}, expectedHistogramModelVector(histograms[len(histograms)-1], nil))
+	queryAndAssert(t, ctx, prom2.Endpoint("http"), func() string { return testHistogramMetricName }, ts, promclient.QueryOptions{}, expectedHistogramModelVector(histograms[len(histograms)-1], nil))
 
-	// Query deduplicated histogram from Thanos Querier.
-	queryAndAssert(t, ctx, querier.Endpoint("http"), func() string { return "fake_histogram" }, time.Now, promclient.QueryOptions{Deduplicate: true}, expectedHistogramModelVector(map[string]string{
-		"prometheus": "prom-ha",
-	}))
+	t.Run("query deduplicated histogram", func(t *testing.T) {
+		queryAndAssert(t, ctx, querier.Endpoint("http"), func() string { return testHistogramMetricName }, ts, promclient.QueryOptions{Deduplicate: true}, expectedHistogramModelVector(histograms[len(histograms)-1], map[string]string{
+			"prometheus": "prom-ha",
+		}))
+	})
 
-	// Query histogram using histogram_count function and deduplication from Thanos Querier.
-	queryAndAssert(t, ctx, querier.Endpoint("http"), func() string { return "histogram_count(fake_histogram)" }, time.Now, promclient.QueryOptions{Deduplicate: true}, model.Vector{
-		&model.Sample{
-			Value: 5,
-			Metric: model.Metric{
-				"foo":        "bar",
-				"prometheus": "prom-ha",
+	t.Run("query histogram using histogram_count fn and deduplication", func(t *testing.T) {
+		queryAndAssert(t, ctx, querier.Endpoint("http"), func() string { return fmt.Sprintf("histogram_count(%v)", testHistogramMetricName) }, ts, promclient.QueryOptions{Deduplicate: true}, model.Vector{
+			&model.Sample{
+				Value: 34,
+				Metric: model.Metric{
+					"foo":        "bar",
+					"prometheus": "prom-ha",
+				},
 			},
-		},
+		})
 	})
 
-	// Query histogram using group function to test pushdown.
-	queryAndAssert(t, ctx, querier.Endpoint("http"), func() string { return "group(fake_histogram)" }, time.Now, promclient.QueryOptions{Deduplicate: true}, model.Vector{
-		&model.Sample{
-			Value:  1,
-			Metric: model.Metric{},
-		},
+	t.Run("query histogram using group function for testing pushdown", func(t *testing.T) {
+		queryAndAssert(t, ctx, querier.Endpoint("http"), func() string { return fmt.Sprintf("group(%v)", testHistogramMetricName) }, ts, promclient.QueryOptions{Deduplicate: true}, model.Vector{
```

Comment:
I think we forgot to enable the pushdown feature flag in `querier` :smile:

Reply:
You are right. I enabled it for this querier.